data_steward/tools/snapshot_by_query.py

```diff
@@ -37,33 +37,64 @@ def create_empty_cdm_tables(snapshot_dataset_id):
     cdm.create_vocabulary_tables(snapshot_dataset_id)
 
 
-def get_field_cast_expr(field, data_type):
+def get_field_cast_expr(field, data_type, required_mode, existing_fields):
```

Comment:
Please accept the entire field object (named something like `src_field`) here rather than three separate params and indicate in docstring the type. Rename the last parameter something like `dst_fields`.